pkg/service/roommanager.go

```diff
@@ -253,10 +253,13 @@ func (r *RoomManager) StartSession(ctx context.Context, roomName string, pi rout
 	opts := rtc.ParticipantOptions{
 		AutoSubscribe: pi.AutoSubscribe,
 	}
-	if err := room.Join(participant, &opts, r.iceServersForRoom(room.Room)); err != nil {
+	if err = room.Join(participant, &opts, r.iceServersForRoom(room.Room)); err != nil {
 		logger.Errorw("could not join room", err)
 		return
 	}
+	if err = r.roomStore.StoreParticipant(ctx, roomName, participant.ToProto()); err != nil {
+		logger.Errorw("could not store participant", err)
+	}
 
 	r.telemetry.ParticipantJoined(ctx, room.Room, participant.ToProto())
 	participant.OnClose(func(p types.Participant) {
```

Comment:
should it similarly delete the participant prior to triggering telemetry?

Reply:
It was deleting before `participant.Close` in `room.OnParticipantChanged`, but the flow wasn't obvious - moved to `participant.onClose` for clarity